packages/form-builder/src/blocks/fields/amount/Edit.tsx

```diff
@@ -1,51 +1,201 @@
-import {__} from '@wordpress/i18n';
+import {__, _n} from '@wordpress/i18n';
 
 import LevelGrid from './level-grid';
 import LevelButton from './level-buttons';
 import Inspector from './inspector';
 import {CurrencyControl, formatCurrencyAmount} from '../../../common/currency';
 import {createInterpolateElement} from '@wordpress/element';
+import {RadioControl} from '@wordpress/components';
+import Notice from './notice';
+import {useState} from 'react';
+
+const periodLookup = {
+    day: {
+        singular: __('day', 'give'),
+        plural: __('days', 'give'),
+        adjective: __('Daily', 'give'),
+    },
+    week: {
+        singular: __('week', 'give'),
+        plural: __('weeks', 'give'),
+        adjective: __('Weekly', 'give'),
+    },
+    month: {
+        singular: __('month', 'give'),
+        plural: __('months', 'give'),
+        adjective: __('Monthly', 'give'),
+    },
+    quarter: {
+        singular: __('quarter', 'give'),
+        plural: __('quarters', 'give'),
+        adjective: __('Quarterly', 'give'),
+    },
+    year: {
+        singular: __('year', 'give'),
+        plural: __('years', 'give'),
+        adjective: __('Yearly', 'give'),
+    },
+};
 
 const Edit = ({attributes, setAttributes}) => {
-    const {levels, priceOption, setPrice, customAmount} = attributes;
+    const {
+        levels,
+        priceOption,
+        setPrice,
+        customAmount,
+        customAmountMin,
+        customAmountMax,
+        recurringEnabled,
+        recurringDonationChoice,
+        recurringPeriod,
+        recurringBillingInterval,
+        recurringBillingPeriod,
+        recurringBillingPeriodOptions,
+        recurringLengthOfTime,
+        recurringOptInDefault,
+    } = attributes;
```

Comment:
Continuing my comment on default values, we might need to start adding fallbacks when destructuring attributes.  This would become more prominent when we have an existing form and then introduce new features/updates to existing blocks on the form (without a database migration).  A topic to note and discuss as a team.